packages/react-sdk/src/db/style.ts

```diff
@@ -0,0 +1,49 @@
+import type { StyleProperty, StyleValue } from "@webstudio-is/css-data";
+import { SharedStyleValue } from "@webstudio-is/css-data";
+import { type ComponentName, getComponentMeta } from "../components";
+import { z } from "zod";
+
+export type PresetStyleItem = {
+  component: ComponentName;
+  property: StyleProperty;
+  value: StyleValue;
+};
+
+// @todo can't figure out how to make component and property to be enum
+export const zPresetStyleItem = z.object({
+  component: z.string(),
+  property: z.string(),
+  value: SharedStyleValue,
+}) as z.ZodType<PresetStyleItem>;
+
+export const zPresetStyle = z.array(zPresetStyleItem);
+
+export type PresetStyle = z.infer<typeof zPresetStyle>;
+
+export const findMissingPresetStyle = (
```

Comment:
I was quite a bit confused about the purpose of this function, please correct me if I am wrong: this is collecting preset styles for all instances as they are being iterated over in a tree, so that only presets for the instances inside the tree are collected, correct?

Reply:
It find all possible components and by these components finds all preset styles.
This utility will probably go away when all tree data will be normalized.

Reply:
`collectUsedPresetStyle` seems like a better name, works well with `collectUsedComponents`

Reply:
actually you are returning an array of presetStyle, so the name should be `collectUsedPresetStyles`

Reply:
please add a test for this function

Reply:
> actually you are returning an array of presetStyle, so the name should be collectUsedPresetStyles

And array of used preset styles missing in passed list.

> please add a test for this function

Maybe not spend time if will be removed?

Reply:
if its going away for sure then, yeah, unless you can write a higher level test that will stay and can confirm unchanged behavior once you refactor